src/Api/AdminConsole/Controllers/OrganizationsController.cs

```diff
@@ -252,6 +252,12 @@ public async Task Leave(Guid id)
             throw new BadRequestException("Your organization's Single Sign-On settings prevent you from leaving.");
         }
 
+        var managingOrgs = await _userService.GetOrganizationsManagingUserAsync(user.Id);
+        if (_featureService.IsEnabled(FeatureFlagKeys.AccountDeprovisioning) && managingOrgs.Where(o => o.Id == id).Count() > 0)
```

Comment:
🎨 The feature check could be done first so an unnecessary call isn't made. And the `.Where(...).Count() > 0` can be simplified to an `Any(...)`.  Something like this:
```suggestion
if (_featureService.IsEnabled(FeatureFlagKeys.AccountDeprovisioning)
    && (await _userService.GetOrganizationsManagingUserAsync(user.Id)).Any(x => x.Id == id))
```